File: DSA/Gautam_Shyamnani_590013078/Day19Question2.cpp
#include <iostream>
#include <vector>
using namespace std;


int findPeakElement(const vector<int>& nums) {
    int left = 0, right = nums.size() - 1;

    while (left < right) {
        int mid = left + (right - left) / 2;

       
        if (nums[mid] < nums[mid + 1]) {
            left = mid + 1;
        } else {
            
            right = mid;
        }
    }

   
    return left;
}

int main() {
    vector<int> nums = {1, 3, 20, 4, 1, 0};  
    int peakIndex = findPeakElement(nums);
    cout << "Peak found at index: " << peakIndex << " with value: " << nums[peakIndex] << endl;
    return 0;
}

File: DSA/upama_kumari_590014909/day_19_question_2.py
def find_peak_element(nums):
    left, right = 0, len(nums) - 1

    while left < right:
        mid = (left + right) // 2

        if nums[mid] > nums[mid + 1]:
            right = mid
        else:
            left = mid + 1
    return left 

nums = [1, 2, 1, 3, 5, 6, 4]
print(find_peak_element(nums))  
